Directly import `roboy_parser`

josephbirkner opened the pull request:
- Adjusted `SemanticSentenceAnalyser` to use new `SemanticAnalyserInterface` from roboy_parser.
- Moved runtime resources from parser to dialog.

#### TODO:
- [x] Update documentation
- [x] Install parser when installing dialog

Commit:
Adapted dialog to directly included parser.

Commit:
Updated parser submodule checkout.

missxa commented:
@josephbirkner please rebase to devel and solve merge conflicts

missxa commented on src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java:
```diff
@@ -59,100 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
+        String response;
+        SemanticAnalyzerInterface.Result result = semanticAnalyzer.analyze(
+            (String) interpretation.getFeature("sentence"));
+        interpretation.getFeatures().put(Linguistics.TOKENS, result.getTokens());
```
so why are we not handling "(no answer)" responses?
why are you ignoring parser results like followupq?

missxa reviewed (changes requested):
- i'd like to see a status check from our jenkins. please use the job WITHOUT MERGE

- doxygen strings are not updated

- i'm not so sure about the whole resources story. can we at leas make it clear which one belong to the parser? like a dedicated folder or so?

Aevyz committed:
RoslessCalls

Created RosLess Calls Class

Aevyz committed:
Alteration of Memory

All commented out due to testing issues

Aevyz committed:
Memory Update

Aevyz committed:
Git Ignore

Aevyz committed:
Memory is now Rosless

See pull request for more information

Aevyz committed:
Revert Git Ignore to prior Devel state

Aevyz committed:
Added Deprecation

Aevyz committed:
Refractor: Changed Name to Neo4jMemoryOperations

Aevyz committed:
Configuration: Start Memory & Init

Run Memory Main in Main of Dialog System
Made Init Memory Clients optional
Add Config to Start Memory and to Run Client Init

Aevyz committed:
Merge pull request #40 from Roboy/memory-ref-jason-3

Configuration: Start Memory & Init

nbasargin commented on pom.xml:
```diff
@@ -372,6 +374,12 @@
             <version>4.2.1</version>
         </dependency>
 
+        <dependency>
```
So, parser is going to be a maven package?

nbasargin commented on src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java:
```diff
@@ -59,100 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
+        interpretation.getFeatures().put(Linguistics.SENTIMENT, result.getSentiment());
+        interpretation.getFeatures().put(Linguistics.POSTAGS, result.getPostags());
+        interpretation.getFeatures().put(Linguistics.LEMMAS, result.getLemmaTokens());
+        return interpretation;
```
Do you plan to merge Wagram's Interpretation type-safety refactoring later?

nbasargin commented on start_submodules.sh:
```diff
@@ -1,28 +0,0 @@
-# start memory
```
no start_submodules.sh anymore? cool

Commit:
Moved parser resources to separate subfolder.

Commit:
Merge branch 'devel' of https://github.com/Roboy/roboy_dialog into joseph-roboy-parser-api

Commit:
Adjusted SemanticParserAnalyzer to new Linguistics API

Commit:
Updated parser submodule checkout

Commit:
Added roboy_parser submodule to roboy_dialog pom.xml.

Commit:
Updated submodule checkout.

Commit:
Fixed memory submodule checkout

Commit:
Updated dialog sumodule checkout.

Commit:
Fixed Parser dependency type.

Aevyz committed:
Comments and Code Quality Changes @nbasargin

Changes according to review

Aevyz committed:
More Code Quality Updates

Aevyz committed:
Updated POM for Jenkins

Aevyz committed:
Update POM for Jenkins

Aevyz committed:
Corrected POM File for use in Intellij

Intellij no longer breaks, I think...

Aevyz committed:
Updated Comments

Aevyz committed:
Removing Unneeded Code

Aevyz committed:
Memory Integration Tests --> No Update Test Yet

Update Test is currently broken

Commit:
Merge branch 'memory-ref-jason-2' of https://github.com/Roboy/roboy_dialog into joseph-roboy-parser-api

Aevyz committed:
Documentation of New Class

Korn \m/

Aevyz committed:
Integration Tests: Working Update Test

Bit of a hacky solution, but it will work for testing.

Commit:
Fixed dialog multimodule layout

Commit:
Fixed parser runtime resource loading

Aevyz committed:
Test Name Consistency

Aevyz commented:
May want to consider adding https://github.com/Roboy/roboy_dialog/pull/39/commits/61e3b07011394276baabfaff7b47b18da7b9af43 to the branch --> Documentation

Aevyz committed:
Neo4j Update Actually Working

Considering my commit names, I'd fit right in with PHP development (see iso 8601 @ http://php.net/manual/en/class.datetime.

Aevyz committed:
Added/Updated Test Fail Messages

Aevyz committed:
Fixed Null Point Exception in Update Test

did not impact the test, but hey, the test looks cleaner ^^

Aevyz committed:
Code Cleanup

Aevyz committed:
Partially standardized Error Msgs

missxa commented on roboy_dialog/src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java.orig:
```diff
@@ -59,6 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
      */
     @Override
     public Interpretation analyze(Interpretation interpretation) {
+<<<<<<< HEAD
```
merge conflict

josephbirkner replied:
A .orig file that slipped into the commit... I'll remove it

josephbirkner committed:
Updated folder names and documentation.
